RiotSwiftUI/Modules/Authentication/VerifyMsisdn/Coordinator/AuthenticationVerifyMsisdnCoordinator.swift

```diff
@@ -212,9 +207,12 @@ final class AuthenticationVerifyMsisdnCoordinator: Coordinator, Presentable {
             authenticationVerifyMsisdnViewModel.displayError(.mxError(mxError.error))
             return
         }
-        
-        // TODO: Handle another other error types as needed.
-        
-        authenticationVerifyMsisdnViewModel.displayError(.unknown)
+
+        switch error {
+        case RegistrationError.invalidPhoneNumber:
+            authenticationVerifyMsisdnViewModel.displayError(.invalidPhoneNumber)
+        default:
+            authenticationVerifyMsisdnViewModel.displayError(.unknown)
+        }
```

Comment:
For some reason I'd never twigged you can simply switch on an error like this without doing an `as? RegistrationError` 🤩